### PR TITLE
Doc: Bluetooth: Mesh: Added shell remote provisioning documentation

### PR DESCRIPTION
**Pull request by Balaklaka**
Add description all shell commands related to remote provisioning.

### CODE REVIEW

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 1]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
 	Abort the Opcodes Aggregator Sequence message. This command clears the Opcodes Aggregator Client context.
 
 
+Remote Provisioning Client
+--------------------------
+
+The Remote Provisioning client is an optional Bluetooth mesh model that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_RPR_CLI` configuration option. The Remote Provisioning Client model is used to support the functionality of dispatching a sequence of access layer messages to nodes supporting the Remote Provisioning Server model.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1481]:
```suggestion
The Remote Provisioning Client is an optional Bluetooth mesh model that can be enabled through the :kconfig:option:`CONFIG_BT_MESH_RPR_CLI` configuration option. The Remote Provisioning Client model supports the functionality of dispatching a sequence of access layer messages to nodes supporting the Remote Provisioning Server model.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 2]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	* ``timeout in seconds``: Scan timeout in seconds. Must be at least 1 second.
+	* ``UUID``: Device UUID to scan for, or NULL to report all devices.
+
+``mesh models rpr scan-ext <timeout in seconds> <UUID> [<AD-type> ... ]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Start extended scanning for unprovisioned devices.
+
+	* ``timeout in seconds``: Scan timeout in seconds. Valid values from @ref BT_MESH_RPR_EXT_SCAN_TIME_MIN to @ref BT_MESH_RPR_EXT_SCAN_TIME_MAX.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1496]: Is there a reason for having @ref before the two values?
Are these config options, could it be written :kconfig:option: instead?
If not, enclose with ``.

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 3]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Start extended scanning for the remote server.
+
+	* ``AD-type``: List of AD types to include in the scan report. Must contain 1 to :kconfig:option:`CONFIG_BT_MESH_RPR_AD_TYPES_MAX` entries.
+
+``mesh models rpr scan-caps``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Get scanning capabilities of Remote Provisioning Server.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1510]:
```suggestion
	Get the scanning capabilities of the Remote Provisioning Server.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 4]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	Start scanning for unprovisioned devices.
+
+	* ``timeout in seconds``: Scan timeout in seconds. Must be at least 1 second.
+	* ``UUID``: Device UUID to scan for, or NULL to report all devices.
+
+``mesh models rpr scan-ext <timeout in seconds> <UUID> [<AD-type> ... ]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Start extended scanning for unprovisioned devices.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1494]:
```suggestion
	Start the extended scanning for unprovisioned devices.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 5]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	* ``timeout in seconds``: Scan timeout in seconds. Valid values from @ref BT_MESH_RPR_EXT_SCAN_TIME_MIN to @ref BT_MESH_RPR_EXT_SCAN_TIME_MAX.
+	* ``UUID``: Device UUID to start extended scanning for.
+	* ``AD-type``: List of AD types to include in the scan report. Must contain 1 to :kconfig:option:`CONFIG_BT_MESH_RPR_AD_TYPES_MAX` entries.
+
+``mesh models rpr scan-srv [<AD-type> ... ]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Start extended scanning for the remote server.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1503]: Could we say Remote Provisioning Server here as well?
```suggestion
	Start the extended scanning for the Remote Provisioning Server.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 6]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+``mesh models rpr scan-caps``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Get scanning capabilities of Remote Provisioning Server.
+
+``mesh models rpr scan-get``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Get current scanning state of Remote Provisioning Server.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1515]:
```suggestion
	Get the current scanning state of the Remote Provisioning Server.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 7]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+``mesh models rpr link-get``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Get the current link status of the Remote Provisioning Server.
+
+``mesh models rpr link-close``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Close any open link on the Remote Provisioning Server.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1530]:
```suggestion
	Close any open links on the Remote Provisioning Server.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 8]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+``mesh models rpr provision-remote <UUID> <NetKeyIndex> <addr>``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Provision a Mesh Node using PB-Remote
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1535]:
```suggestion
	Provision a mesh node using the PB-Remote provisioning bearer.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 9]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	* ``UUID``: UUID of the unprovisioned node.
+	* ``NetKeyIndex``: Network Key Index to give to the unprovisioned node.
+	* ``addr``: Address to assign to remote device. If addr is 0, the lowest available address will be chosen.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1539]:
```suggestion
	* ``addr``: Address to assign to the remote device. If ``addr`` is 0, the lowest available address will be chosen.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 10]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+``mesh models rpr reprovision-remote <addr> [<comp changed: false, true>]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Reprovision a Mesh Node using PB-Remote
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1544]:
```suggestion
	Reprovision a mesh node using the PB-Remote provisioning bearer.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 11]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	* ``addr``: Address to assign to remote device. If addr is 0, the lowest available address will be chosen.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1546]:
```suggestion
	* ``addr``: Address to assign to the remote device. If ``addr`` is 0, the lowest available address will be chosen.
```

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 12]
```diff
@@ -1475,6 +1475,90 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+	* ``comp changed``: The target node has indicated that its composition data has changed. Note that the target node will reject the update if this isn't true.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1547]:
```suggestion
	* ``comp changed``: The Target node has indicated that its Composition Data has changed. Note that the Target node will reject the update if this isn't true.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/shell.rst (hunk 2 above)
Comment (Balaklaka) [reply doc/connectivity/bluetooth/api/mesh/shell.rst:1496]: These are not kconfig options.

File: doc/connectivity/bluetooth/api/mesh/shell.rst (hunk 10 above)
Comment (Balaklaka) [reply doc/connectivity/bluetooth/api/mesh/shell.rst:1544]: Will not be better to say "the Remote Provisioning Bearer (PB-Remote)" at the end.

**Review by akredalen — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/shell.rst (hunk 10 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/shell.rst:1544]: If you'd like :)
As long as you write "the remote provisioning bearer" instead (no title casing, at least not in the spec).

**Review by akredalen — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/shell.rst [hunk 13]
```diff
@@ -1475,6 +1475,92 @@ The Opcodes Aggregator client is an optional Bluetooth mesh model that can be en
...
+
+	* ``timeout in seconds``: Scan timeout in seconds. Must be at least 1 second.
+	* ``UUID``: Device UUID to scan for. If omitted, all devices will be reported.
+
+``mesh models rpr scan-ext <timeout in seconds> <UUID> [<AD-type> ... ]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+	Start the extended scanning for unprovisioned devices.
+
+	* ``timeout in seconds``: Scan timeout in seconds. Valid values from ``@ref BT_MESH_RPR_EXT_SCAN_TIME_MIN`` to ``@ref BT_MESH_RPR_EXT_SCAN_TIME_MAX``.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/shell.rst:1498]: Do we need @ref or could it be omitted?

**Review by PavelVPV — APPROVED** [review]

**Review by akredalen — APPROVED** [review]

File: doc/connectivity/bluetooth/api/mesh/shell.rst (hunk 13 above)
Comment (Balaklaka) [reply doc/connectivity/bluetooth/api/mesh/shell.rst]: I change this to :c:macro: and that will show this in a special way in the documentation.

**Review by mia-ko — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Pull request merged**